### PR TITLE
refactor: update datasources to use TS files instead of JSON files

### PR DESCRIPTION
**Pull request by mrmodise**
Update datasources to use TS files config instead of JSON files. Part of https://github.com/strongloop/loopback-next/issues/5220

### CODE REVIEW

**Comment by mrmodise** [comment]
@jannyHou @hacksparrow pipelines failing:

Appveyor:

`         returns product recommendations for a user using gRPC:
     Error: 14 UNAVAILABLE: No connection established
      at Object.callErrorFromStatus (packages\shopping\node_modules\@grpc\grpc-js\src\call.ts:81:24)
`

Travis:
`The command "commitlint-travis" exited with 1.`

Interesting that locally none of these are failing. Any advise?

**Review by bajtos — CHANGES_REQUESTED** [review]
Thank you @mrmodise for the pull request! ❤️ 

The code changes look good to me, can you remove `.datasource.json` files too? (`git rm`)

**Comment by bajtos** [comment]
> The command "commitlint-travis" exited with 1.

It complains about commit messages not being signed off, which is weird. We will have to investigate.

**Comment by bajtos** [comment]
@raymondfeng @hacksparrow @jannyHou can you please help with troubleshooting test failures? I am not very familiar with this code base and the testing infrastructure in use.

**Comment by bajtos** [comment]
@mrmodise can you please rebase your pull request on top of the latest `master`?

**Comment by mrmodise** [comment]
> Thank you @mrmodise for the pull request! ❤️
> 
> The code changes look good to me, can you remove `.datasource.json` files too? (`git rm`)

Welcome. Pushed the requested changes

**Comment by mrmodise** [comment]
> The CI failure message is:
> 
> ```
>   1) UserController
>        user product recommendation (service) api
>          returns product recommendations for a user using gRPC:
>      Error: 14 UNAVAILABLE: No connection established
>       at Object.callErrorFromStatus (packages/shopping/node_modules/@grpc/grpc-js/src/call.ts:81:24)
>       at Object.onReceiveStatus (packages/shopping/node_modules/@grpc/grpc-js/src/client.ts:334:36)
>       at Object.onReceiveStatus (packages/shopping/node_modules/@grpc/grpc-js/src/client-interceptors.ts:434:34)
>       at Object.onReceiveStatus (packages/shopping/node_modules/@grpc/grpc-js/src/client-interceptors.ts:397:48)
>       at process.nextTick (packages/shopping/node_modules/@grpc/grpc-js/src/call-stream.ts:237:24)
>       at process._tickCallback (internal/process/next_tick.js:61:11)
> ```
> 
> @mrmodise is it passing locally? @raymondfeng any idea?

Correct, its passing locally

File: packages/shopping/src/datasources/recommender-grpc.datasource.ts [hunk 1]
```diff
@@ -7,7 +7,12 @@ import {inject} from '@loopback/core';
 import {juggler, AnyObject} from '@loopback/repository';
 import path from 'path';
 
-const config = require('./recommender-grpc.datasource.config.json');
+const config = {
+  name: 'recommender_grpc',
+  connector: 'loopback-connector-grpc',
+  url: '127.0.0.1:50051',
```
Comment (jannyHou) [inline packages/shopping/src/datasources/recommender-grpc.datasource.ts:13]: The old config has port as 50000, I think it breaks the test.

`"url": "127.0.0.1:50000",`

**Review by jannyHou — CHANGES_REQUESTED** [review]
@mrmodise Appreciate your contribution again. Please check the grpc server's port number as I commented. I suspect it causes the failure.

File: packages/shopping/src/datasources/recommender-grpc.datasource.ts (hunk 1 above)
Comment (mrmodise) [reply packages/shopping/src/datasources/recommender-grpc.datasource.ts]: aha! updated accordingly

**Comment by bajtos** [comment]
> Merge branch 'master' into update-datasource-style

Please note we are not using merge commits, we rebase on top of the new master instead. See https://loopback.io/doc/en/lb4/submitting_a_pr.html#6-rebasing

The build is mostly passing now with the exception of `commit-lint`. I think the problem is likely to go away after you rebase. Can you also squash all commits into a single one please?

See https://loopback.io/doc/en/lb4/submitting_a_pr.html#9-final-rebase-and-squashing-of-commits

**Review by bajtos — APPROVED** [review]
Code changes LGTM, let's clean up the git history now.

**Comment by mrmodise** [comment]
> > Merge branch 'master' into update-datasource-style
> 
> Please note we are not using merge commits, we rebase on top of the new master instead. See https://loopback.io/doc/en/lb4/submitting_a_pr.html#6-rebasing
> 
> The build is mostly passing now with the exception of `commit-lint`. I think the problem is likely to go away after you rebase. Can you also squash all commits into a single one please?
> 
> See https://loopback.io/doc/en/lb4/submitting_a_pr.html#9-final-rebase-and-squashing-of-commits

Very helpful. I updated the changes accordingly. Thanks @bajtos

**Review by jannyHou — APPROVED** [review]
👍

**Review by raymondfeng — APPROVED** [review]

**Pull request merged**